cni/network/network.go

```diff
@@ -322,6 +323,63 @@
 	log.Printf("Released lock file")
 	return nil
 }
+
+// MigrateKeyValueStore attempts to idempotently migrate a kvstore at the
+// legacy location in to the new location as a one-time operation. If a
+// migration is not necessary, returns nil. If a migration is required but
+// does not succeed for any reason, returns an error.
+func (plugin *netPlugin) MigrateKeyValueStore() error {
+	filename := platform.CNIRuntimePath + plugin.Name + ".json"
+	legacyFilename := platform.LegacyCNIRuntimePath + plugin.Name + ".json"
+
+	if plugin.Store == nil {
+		return errors.New("kvstore is not initialized")
+	}
+
+	// test if the backing state file exists in the current location
+	// if it does, no migration is necessary
+	ok, err := platform.CheckIfFileExists(filename)
+	if err != nil {
+		return fmt.Errorf("error checking current file store location %s: %w", filename, err)
+	}
+	if ok {
+		return nil
+	}
+
+	// a current statefile does not exists, so we need to check the legacy
+	// location to see if there is one there to migrate
+	// if it does not, no migration is necessary
+	ok, err = platform.CheckIfFileExists(legacyFilename)
+	if err != nil {
+		return fmt.Errorf("error checking legacy file store location %s: %w", legacyFilename, err)
+	}
+	if !ok {
+		return nil
+	}
+
+	// there is no current statefile, but there is a legacy one
+	// attempt to migrate
+
+	// take the lock
+	if err := plugin.acquireLockForStore(); err != nil {
+		return fmt.Errorf("error locking store for migration: %w", err)
+	}
+	defer plugin.releaseLockForStore()
+
+	// check the current location again in case someone else wrote it
+	// before we got the lock
+	ok, err = platform.CheckIfFileExists(filename)
+	if err != nil {
+		return fmt.Errorf("error checking current file store location %s after lock: %w", filename, err)
+	}
+	if ok {
+		return nil
+	}
+
+	// we have the lock and still require a migration
+	return platform.Copy(legacyFilename, filename)
```

Comment:
@tamilmani1989, this wont impact the temp file created, right?

Reply:
nope it wont